app/src/main/java/com/github/sdp/mediato/ui/viewmodel/ExploreViewModel.java

```diff
@@ -34,7 +35,7 @@ public ExploreViewModel(@NonNull Application application) {
      * Sets the username of the user who is currently logged in and generates the list of posts
      * @param username the username of the user who is currently logged in
      */
-    public void setUsername(String username){
+    public void setData(String username) {
         this.username = username;
         createNearbyUsersPosts();
     }

```

Comment:
Why did you change the name of the function with Data and not Username when this function is to set the username?

Reply:
I tried to pass it other data at some point then I forgot to change it back to just username. I fixed that.